File: 103-2_iOS程式設計/CoreMotionSample02/CoreMotionSample02/ViewController.m
//
//  ViewController.m
//  CoreMotionSample02
//
//  Created by CSIEVNUT403 on 2015/6/8.
//  Copyright (c) 2015年 403. All rights reserved.
//

#import "ViewController.h"
#import <CoreMotion/CoreMotion.h>

@interface ViewController ()

@property (strong, nonatomic) IBOutlet UILabel *gyroX;
@property (strong, nonatomic) IBOutlet UILabel *gyroY;
@property (strong, nonatomic) IBOutlet UILabel *gyroZ;

@property (strong, nonatomic) CMMotionManager *motionManager;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.motionManager = [[CMMotionManager alloc] init];
    self.motionManager.gyroUpdateInterval = 0.2;
    
    [self.motionManager startGyroUpdatesToQueue:[NSOperationQueue currentQueue] withHandler:^(CMGyroData *gyroData, NSError *error) {
        
        [self outputRotationData:gyroData.rotationRate];
        
        if (error) {
            NSLog(@"%@", error);
        }
    }];
}

- (void)outputRotationData:(CMRotationRate)rotation {
    self.gyroX.text = [NSString stringWithFormat:@"%f r/s", rotation.x];
    self.gyroY.text = [NSString stringWithFormat:@"%f r/s", rotation.y];
    self.gyroZ.text = [NSString stringWithFormat:@"%f r/s", rotation.z];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: 106-2_iOS程式設計/Label/Label/ViewController.swift
import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        let labelTest:UILabel = UILabel() //建立元件
        labelTest.frame = CGRect(x: 60, y: 80, width: 240, height: 40) //設定位置及大小
        labelTest.text = "這是程式碼建立的元件" //設定顯示文字
        labelTest.textColor = UIColor.redColor() //文字為紅色
        view.addSubview(labelTest) //加入介面
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

